### PR TITLE
Still create lambdas in the cloud layer so we pick up the right cloud-options for them.

### PR DESCRIPTION
**Pull request by CyrusNajmabadi**

### CODE REVIEW

File: aws/api.ts [hunk 1]
```diff
@@ -293,7 +296,15 @@ function convertHandlers(handlers: cloud.RouteHandler[]): lambda.Callback<x.Requ
         next();
     };
 
-    return result;
+    const routeName = name + sha1hash(route.method + ":" + route.path);
+
+    // Create the CallbackFunction in the cloud layer as opposed to just passing the callback down
+    // to pulumi-aws.  This ensures that the right configuration values are used that will
+    // appropriately respect user settings around things like codepaths/policies etc.
```
Comment (swgillespie) [inline aws/api.ts]: Not sure I follow this comment. Maybe I've just lost track of the changes around here, but how does this respect user settings (and what does that mean)?

File: aws/function.ts [hunk 2]
```diff
@@ -27,6 +27,42 @@ export function createFactoryFunction(
...
 }
 
+export function createCallbackFunction(
+        name: string,
+        handler: aws.serverless.Handler | aws.serverless.HandlerFactory,
+        isFactoryFunction: boolean,
+        opts?: pulumi.ResourceOptions): aws.lambda.CallbackFunction<any, any> {
+
+    const policies = [...getComputeIAMRolePolicies()];
+    let vpcConfig: aws.serverless.FunctionOptions["vpcConfig"] | undefined;
```
Comment (swgillespie) [inline aws/function.ts]: not your code, but isn't this a tautology? `T | undefined` == `T`?

Comment (swgillespie) [reply aws/function.ts]: oh, I guess it's not for anything falsy... 🤦‍♂️

Comment (CyrusNajmabadi) [reply aws/function.ts]: yes.  yes it is.  fixing.

File: aws/function.ts [hunk 3]
```diff
@@ -27,6 +27,42 @@ export function createFactoryFunction(
...
+        codePathOptions: {
+            extraIncludePaths: functionIncludePaths,
+            extraIncludePackages: functionIncludePackages,
+        },
+        callback: isFactoryFunction ? undefined : <aws.serverless.Handler>handler,
+        callbackFactory: isFactoryFunction ? <aws.serverless.HandlerFactory>handler : undefined,
+    };
+
+    return new aws.lambda.CallbackFunction(name, args, opts);
+}
```
Comment (CyrusNajmabadi) [inline aws/function.ts]: this code just moved up from below.  It is otherwise unchanged.

**Pull request merged**